### PR TITLE
Fix auto-update of secrets into options

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Comment by MattGal** [comment]
Can you link the issue this relates to please?

**Review by MattGal — APPROVED** [review]

**Review by adiaaida — APPROVED** [review]

**Comment by alexperovich** [comment]
This is a bunch of secret manager changes I've had sitting in my repo for a bit over a week, I wanted to get them out.

**Review by garath — APPROVED** [review]

File: src/Maestro/DependencyUpdater/Program.cs [hunk 1]
```diff
@@ -49,37 +49,29 @@ public static void Configure(IServiceCollection services)
...
             {
-                var config = provider.GetRequiredService<IConfiguration>();
-                IConfigurationSection section1 = config.GetSection("GitHub");
-                section1.Bind(options);
+                s.Bind(o);
             });
             services.AddGitHubTokenProvider();
 
             services.AddAzureDevOpsTokenProvider();
-            services.Configure<AzureDevOpsTokenProviderOptions>((options, provider) =>
```
Comment (ChadNedzlek) [inline src/Maestro/DependencyUpdater/Program.cs:61]: Is there anything we can do to prevent people from accidentally calling the "bad" one?

**Review by ChadNedzlek — APPROVED** [review]

File: src/Maestro/DependencyUpdater/Program.cs (hunk 1 above)
Comment (alexperovich) [reply src/Maestro/DependencyUpdater/Program.cs:61]: I can't think of anything other than an analyzer. There is a possibility I could write something that would make calls to the bad one register as ambiguous, but that would be really hacky.

**Comment by riarenas** [comment]
This doesn't seem to have been mirrored and hence didn't trigger an official build. I'll see if I can find what happened

**Comment by riarenas** [comment]
Found it: https://dev.azure.com/dnceng/internal/_build/results?buildId=1238416&view=logs&j=a3453f18-ed27-56b8-78ee-81a7631275fb&t=1ec614b8-726e-5075-1cd1-dd16c6e99f29&l=32. We merged a PR in the mirror.

@ulisesh for Arcade-services, github is the source of truth that gets mirrored to azure devops. https://dev.azure.com/dnceng/internal/_git/dotnet-arcade-services/pullrequest/16614 breaks the mirroring as it's a commit that is only present in azure devops. I'm going to force push github main into the azure devops repo to unblock this again, but it means your commit will be lost. I created https://github.com/dotnet/arcade-services/pull/1676 to make sure the commit makes it through for the next rollout.

**Pull request merged**